### PR TITLE
feat: Fix crash and add toast, permission 

### PR DESCRIPTION
**Pull request by weiraneve**
## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist
1. [ ] Makes your submission pass tests?
2. [ ] Have you lint your code locally before submission?
3. [ ] Have you updated the wiki of your change?

## Reference

### CODE REVIEW

File: core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt [hunk 1]
```diff
@@ -84,14 +84,16 @@ class FileManager : StorageInterface {
...
+            if (!exists(filename)) {
+                createFile(filename)
+            }
+            try {
+                val file = File(path, filename)
+                file.writeText(content)
+            } catch (e: IOException) {
+                Logger.e("$WRITE_TEXT_FILE_EXCEPTION $e")
+            }
         }
```
Comment (Tony---Zhang) [inline core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt:99]: Else

File: core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt [hunk 2]
```diff
@@ -84,14 +84,16 @@ class FileManager : StorageInterface {
     }
 
     override fun writeTextToFile(filename: String, content: String) {
-        if (!exists(filename)) {
-            createFile(filename)
-        }
-        try {
-            val file = File(path, filename)
-            file.writeText(content)
-        } catch (e: IOException) {
```
Comment (Tony---Zhang) [inline core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt:93]: catch的内容不够，之前直接崩溃

File: core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt (hunk 1 above)
Comment (weiraneve) [reply core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt:99]: get it

File: core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt (hunk 2 above)
Comment (weiraneve) [reply core-storage/src/main/java/com/thoughtworks/ark/core/storage/FileManager.kt:93]: Get it and I add permission to it.

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt [hunk 3]
```diff
@@ -68,12 +76,24 @@ class StorageViewModel @Inject constructor(private val fileManager: StorageInter
         }
     }
 
+    private fun showError(context: Context) {
```
Comment (Tony---Zhang) [inline sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: We have `Context.showToast` in `ContextExtensions.kt`

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt [hunk 4]
```diff
@@ -37,7 +39,9 @@ class StorageViewModel @Inject constructor(private val fileManager: StorageInter
     private val defaultWriteContent = "demo content"
 
     init {
-        fileManager.path += defaultPath
+        if (fileManager.path != null) {
```
Comment (Tony---Zhang) [inline sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: if let

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt [hunk 5]
```diff
@@ -50,8 +54,12 @@ class StorageViewModel @Inject constructor(private val fileManager: StorageInter
         }
     }
 
-    private fun writeFile() {
-        fileManager.writeTextToFile(defaultFilename, defaultWriteContent)
+    private fun writeFile(context: Context) {
+        if (fileManager.path == null) {
```
Comment (Tony---Zhang) [inline sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: `isNullOrBlank()`

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt (hunk 3 above)
Comment (weiraneve) [reply sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: get it

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt (hunk 4 above)
Comment (weiraneve) [reply sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: get it

File: sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt (hunk 5 above)
Comment (weiraneve) [reply sample/src/main/java/com/thoughtworks/ark/sample/storage/StorageViewModel.kt]: get it

**Review by Knight912 — APPROVED** [review]

**Review by jie-meng — APPROVED** [review]

**Pull request merged**